src/main/java/com/google/impactdashboard/database_manager/data_read/DataReadManagerImpl.java

```diff
@@ -114,6 +114,27 @@ public double getAverageIAMBindingsInPastYear(String projectId) {
     return datesToBindings;
   }
 
+  /**
+   * Queries the IAM Table and returns the most recent timestamp, or 0 if there is
+   * no data.
+   */
+  public long getMostRecentTimestamp() {
+    QueryJobConfiguration queryConfiguration = queryConfigurationBuilder
+      .getMostRecentTimestampConfiguration()
+      .build();
+    TableResult results = database.readDatabase(queryConfiguration);
+
+    long maxTimestamp = 0;
+    try {
+      for (FieldValueList row : results.iterateAll()) {
+        maxTimestamp = row.get("Max_Timestamp").getTimestampValue() / 1000;
+      }
+    } catch (NullPointerException np) {
```

Comment:
generally speaking, exceptions are slower than statements. 

consider doing an if statement instead of relying on exceptions so much :)

Reply:
The NullPointerException is thrown by getTimestampValue if the row result is null (there was no data in the table). Is there a way to handle this with if statements?

Reply:
NullPointerException is thrown isNull() returns true. So can call isNull() on the FieldValue first.

Reply:
I see! Thanks so much I've fixed it!